File: 0013-roman-to-integer/0013-roman-to-integer.c
int romanToInt(char* s) {
    int myMap[128];

    myMap['I'] = 1;
    myMap['V'] = 5;
    myMap['X'] = 10;
    myMap['L'] = 50;
    myMap['C'] = 100;
    myMap['D'] = 500;
    myMap['M'] = 1000;

    int result = myMap[s[strlen(s) - 1]];
    for (int i = strlen(s) - 2; i >= 0; --i) {
        if (myMap[s[i]] < myMap[s[i + 1]]) {
            result -= myMap[s[i]];
        } else {
            result += myMap[s[i]];
        }
    }
    return result;
}

File: 0013-roman-to-integer/0013-roman-to-integer.ts
function romanToInt(s: string): number {
    const myMap: { [key: string]: number } = {
        'I': 1,
        'V': 5,
        'X': 10,
        'L': 50,
        'C': 100,
        'D': 500,
        'M': 1000
    };

    let result: number = myMap[s.charAt(s.length - 1)];

    for (let i: number = s.length - 2; i >= 0; --i) {
        if (myMap[s.charAt(i)] < myMap[s.charAt(i + 1)]) {
            result -= myMap[s.charAt(i)];
        } else {
            result += myMap[s.charAt(i)];
        }
    }

    return result;
};